cacheutil/cacheserver.go

```diff
@@ -23,38 +22,62 @@ type IncomingBuffer struct {
 
 //IncomingDataCache cache server converts it into this
 type IncomingDataCache struct {
-	hosts map[string]*ShardedIncomingDataCache
-	lock  *sync.RWMutex
+	hosts  map[string]*ShardedIncomingDataCache
+	maxTTL int64
+	lock   *sync.RWMutex
 }
 
 //ShardedIncomingDataCache types of sharded cache collectd, influxdb etc
 //ShardedIncomingDataCache  ..
 type ShardedIncomingDataCache struct {
-	plugin map[string]incoming.DataTypeInterface
-	lock   *sync.RWMutex
+	plugin     map[string]incoming.DataTypeInterface
+	lastAccess int64
+	maxTTL     int64
+	lock       *sync.RWMutex
 }
 
 //NewCache   .. .
-func NewCache() IncomingDataCache {
+func NewCache(maxttl int64) IncomingDataCache {
+	if maxttl == 0 {
+		maxttl = MAXTTL
+	}
 	return IncomingDataCache{
-		hosts: make(map[string]*ShardedIncomingDataCache),
-		lock:  new(sync.RWMutex),
+		hosts:  make(map[string]*ShardedIncomingDataCache),
+		maxTTL: maxttl,
+		lock:   new(sync.RWMutex),
 	}
 }
 
 //NewShardedIncomingDataCache   .
-func NewShardedIncomingDataCache() *ShardedIncomingDataCache {
+func NewShardedIncomingDataCache(maxttl int64) *ShardedIncomingDataCache {
 	return &ShardedIncomingDataCache{
 		plugin: make(map[string]incoming.DataTypeInterface),
+		maxTTL: maxttl,
 		lock:   new(sync.RWMutex),
 	}
+
+}
+
+//FlushAll Flush raw meterics data
+func (i *IncomingDataCache) FlushAll() {
+	allHosts := i.GetHosts()
+	for key, plugin := range allHosts {
+		//fmt.Fprintln(w, hostname)
```

Comment:
You can delete this

Reply:
I need this for my test case. I check my cache clearing algorithm using this..at least for now